### ISSUE TITLE
Iterable and dictionary unpacking in comprehension should be syntax error

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1002]
**Describe the bug**
Pyright allows iterable and dictionary unpacking in comprehensions of the form `(*i for i in [])`.  This is a syntax error in Python - iterables and dictionaries cannot be unpacked inside comprehensions.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That it would be recognised as a syntax error / fail to parse.

**Screenshots or Code**
```py
(*i for i in [])
[*i for i in []]
{**d for d in []}
```

**VS Code extension or command-line**
Extension, 1.1.66

**Comment by erictraut** [comment]
Thanks for reporting this. It will be fixed in the next release of Pyright.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.67, which I just published. It will also be included in next week's release of Pylance.
